metaflow/task.py

```diff
@@ -318,6 +318,8 @@ def run_step(self,
         })
         logger = self.event_logger
         start = time.time()
+        self.metadata.start_task_heartbeat(self.flow.name, run_id, step_name,
```

Comment:
Nit: why here and not with the other side cars (logger below)?

Reply:
it's how we do it internally. IIRC we didn't want the heartbeat sidecar within the try/except/finally block because in case of failure we wanted different behavior. But given your previous comment here:
https://github.com/Netflix/metaflow/pull/333#discussion_r625990515

If we decide to start treating all heartbeat failures as soft failures then that's really no longer a concern and we can move it 2 lines down right above the logger